source/slang/slang-ir-autodiff-loop-analysis.h

```diff
@@ -0,0 +1,181 @@
+// slang-ir-autodiff-loop-analysis.h
+#pragma once
+
+#include "slang-ir-autodiff-region.h"
+#include "slang-ir-autodiff.h"
+#include "slang-ir-dominators.h"
+#include "slang-ir-insts.h"
+#include "slang-ir.h"
+
+namespace Slang
+{
+struct SimpleRelation
```

Comment:
Can we handle the case where the right hand side of a relation is another variable and not just a constant?

Reply:
Yes, it should be possible to extend the current system. 
Basically, we need a way to do 'substitution' on both sides as we move propagate through the blocks and phi parameters have to be translated into their args. Right now, we assume that each relation references only one inst.

I think for now, this covers a lot of cases, but I agree that we should extend this to support those cases too. Preferably in a follow-up PR, since this one has gotten rather large?